src/Core/Form/ChoiceProvider/OrderStateByIdChoiceProvider.php

```diff
@@ -70,4 +81,22 @@ public function getChoices()
 
         return $choices;
     }
+
+    /**
+     * Get order state choices attributes.
+     *
+     * @return array
+     */
+    public function getChoicesAttributes()
+    {
+        $orderStates = $this->orderStateDataProvider->getOrderStates($this->languageId);
+        $attrs = [];
+
+        foreach ($orderStates as $orderState) {
+            $attrs[$orderState['name']]['data-background-color'] = $orderState['color'];
```

Comment:
Should be better to use an id instead of a name here :thinking:  (`$orderState['name']`)
i'm afraid about having the same name twice

Reply:
@atomiix

Reply:
I'm afraid we can't. The way Symfony works with `select` (and `option`) is that it use the `value` of the array as the `name` property of the `option` element and the `key` of the array as the `option` text. [See here](https://symfony.com/doc/current/reference/forms/types/choice.html#choices)